src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java

```diff
@@ -0,0 +1,348 @@
+        for (int i = 0; i < results.size(); ++i) {
+            if (i > 0) {
+                renderedResults.append(',');
+            }
+            renderedResults.append('\n');
+            renderedResults.append(results.get(i));
+        }
+        final String rendered = report
+            .replace(VERSION_PLACEHOLDER, String.valueOf(version))
+            .replace(RESULTS_PLACEHOLDER, renderedResults.toString());
+        writer.print(rendered);
+        if (closeStream) {
+            writer.close();
+        }
+        else {
+            writer.flush();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * Following idea suppressions are false positives
+     *
+     * @noinspection DynamicRegexReplaceableByCompiledPattern
+     */
+    @Override
+    public void addError(AuditEvent event) {
+        if (event.getColumn() > 0) {
+            results.add(resultLineColumn
+                .replace(SEVERITY_LEVEL_PLACEHOLDER, renderSeverityLevel(event.getSeverityLevel()))
+                .replace(URI_PLACEHOLDER, event.getFileName())
+                .replace(COLUMN_PLACEHOLDER, Integer.toString(event.getColumn()))
+                .replace(LINE_PLACEHOLDER, Integer.toString(event.getLine()))
+                .replace(MESSAGE_PLACEHOLDER, escape(event.getMessage()))
+                .replace(RULE_ID_PLACEHOLDER, event.getViolation().getKey())
+            );
+        }
+        else {
+            results.add(resultLineOnly
+                .replace(SEVERITY_LEVEL_PLACEHOLDER, renderSeverityLevel(event.getSeverityLevel()))
+                .replace(URI_PLACEHOLDER, event.getFileName())
+                .replace(LINE_PLACEHOLDER, Integer.toString(event.getLine()))
+                .replace(MESSAGE_PLACEHOLDER, escape(event.getMessage()))
+                .replace(RULE_ID_PLACEHOLDER, event.getViolation().getKey())
+            );
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * Following idea suppressions are false positives
+     *
+     * @noinspection DynamicRegexReplaceableByCompiledPattern
+     */
+    @Override
+    public void addException(AuditEvent event, Throwable throwable) {
+        final StringWriter stringWriter = new StringWriter();
+        final PrintWriter printer = new PrintWriter(stringWriter);
+        throwable.printStackTrace(printer);
+        if (event.getFileName() == null) {
+            results.add(resultErrorOnly
+                .replace(SEVERITY_LEVEL_PLACEHOLDER, renderSeverityLevel(event.getSeverityLevel()))
+                .replace(MESSAGE_PLACEHOLDER, escape(stringWriter.toString()))
+            );
+        }
+        else {
+            results.add(resultFileOnly
+                .replace(SEVERITY_LEVEL_PLACEHOLDER, renderSeverityLevel(event.getSeverityLevel()))
+                .replace(URI_PLACEHOLDER, event.getFileName())
+                .replace(MESSAGE_PLACEHOLDER, escape(stringWriter.toString()))
+            );
+        }
```

Comment:
shorter version is
```
String result = resultFileOnly
                .replace(SEVERITY_LEVEL_PLACEHOLDER, renderSeverityLevel(event.getSeverityLevel()))
                .replace(MESSAGE_PLACEHOLDER, escape(stringWriter.toString()));
if (event.getFileName() != null) {
                result = result.replace(URI_PLACEHOLDER, event.getFileName());
}
results.add(result);
```

Reply:
The if block is on `resultErrorOnly` while the else block is on `resultFileOnly`. Therefore it is not possible to shorten as suggested.